### PR TITLE
docs: Add key concepts

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Working with my colleague @tom-halpin on introducing developing/contributing to Backstage and what would be necessary for a team to actually run it, one of the points he made to me was that Backstage expects a lot of concepts to be known to users. Thinking on his points, I'm introducing a key concepts page here. The idea is that to TypeScript/Node developers, a lot of Backstage uses common dev patterns, _however_ those coming from different development backgrounds may find some of the concepts totally new.  This intends to point out some of them, with references to outside placed to learn more, but does NOT intend to explain each of them in any detail. Overtime, this list could simply have multiple useful blog posts added to.

I'm also conscious we don't want it to become a duplicate of the Glossary, but there is overlap I think.

image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/getting-started/concepts.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+id: concepts
+title: Key Concepts
+# prettier-ignore
+description: High level of key concepts used in the Backstage project
```
Comment (freben) [inline docs/getting-started/concepts.md:5]:
```suggestion
description: High level key concepts used in the Backstage project
```

File: docs/getting-started/concepts.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+For new users to Backstage, there are certain concepts which are central to its
```
Comment (freben) [inline docs/getting-started/concepts.md:8]: Well for all users :)

File: docs/getting-started/concepts.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+design and functionality. Being an expert in each of these concepts is not
+necessary, however having a base understanding of each will make administering,
+configuring, and operating Backstage easier to understand.
```
Comment (freben) [inline docs/getting-started/concepts.md:11]:
```suggestion
configuring, and operating Backstage easier.
```

**Comment by freben** [comment]
One more thing. This page could eventually be amended with a key concepts _of Backstage specifically_ too. Catalog/entities, plugins, whatnot. High level picture stuff

**Comment by adamdmharvey** [comment]
Yeah that's a good point, I can see the nuance of pre-req key concepts I guess and core Backstage concepts. I think where some of this was leading is my colleague did a great job in about a week getting Backstage working for himself, but I had to help him with some things which I had learned over the months. We don't use JavaScript at all, no Yarn or monorepo concepts at all through our entire dev org, and so a lot of this was new to him but we're fast learners so wanted to try to pick a lot of this up. But it's sort of these things which if we understand them better, could open backstage's user population even further, and potentially guide us as to where we could short circuit/improve certain parts of the tool overtime I think.

**Review by freben — APPROVED** [review]

**Pull request merged**